src/codegen/general.rs

```diff
@@ -744,22 +744,42 @@ pub fn declare_default_from_new(
     env: &Env,
     name: &str,
     functions: &[analysis::functions::Info],
+    has_builder: bool,
 ) -> Result<()> {
     if let Some(func) = functions.iter().find(|f| {
         !f.visibility.hidden()
             && f.status.need_generate()
             && f.name == "new"
-            && f.parameters.rust_parameters.is_empty()
             // Cannot generate Default implementation for Option<>
             && f.ret.parameter.as_ref().map_or(false, |x| !*x.lib_par.nullable)
     }) {
-        writeln!(w)?;
-        version_condition(w, env, func.version, false, 0)?;
-        writeln!(w, "impl Default for {} {{", name)?;
-        writeln!(w, "    fn default() -> Self {{")?;
-        writeln!(w, "        Self::new()")?;
-        writeln!(w, "    }}")?;
-        writeln!(w, "}}")?;
+        if func.parameters.rust_parameters.is_empty() {
+            writeln!(w)?;
+            version_condition(w, env, func.version, false, 0)?;
+            writeln!(
+                w,
+                "impl Default for {} {{
+                     fn default() -> Self {{
+                         Self::new()
+                     }}
+                 }}",
+                name
+            )?;
+        } else if has_builder {
+            // create an alternative default implementation the uses `glib::object::Object::new()`
+            writeln!(w)?;
+            version_condition(w, env, func.version, false, 0)?;
+            writeln!(
+                w,
+                "impl Default for {} {{
+                     fn default() -> Self {{
+                         match glib::object::Object::new::<Self>(&[]) {{
+                              Ok(obj) => obj,
+                              Err(err) => panic!(\"Can't construct {} object with default parameters: {{}}\", err),
```

Comment:
Why not just `expect()` here instead? :)

Reply:
This can't really fail FWIW. The only way how this could fail is if there's more broken inside the bindings.

Reply:
Because I thought I'd be nice to have the underlying glib error printed in the panic message as well. And `expect()` just takes a string.

Reply:
I like it like this actually. ^^'

Reply:
`expect()` is also printing the underlying error

Reply:
Alright, should I use expect then or is it the current version okay?

Reply:
I'll leave the final decision to @sdroege, I'm fine with both.

Reply:
`expect()` is less code, especially for an error that can't really happen.